packages/react-native-session-replay/ios/Sources/SvgViewRecorder.swift

```diff
@@ -0,0 +1,195 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+import Foundation
+@_spi(Internal)
+import DatadogSessionReplay
+import DatadogInternal
+import DatadogSDKReactNative
+import UIKit
+import React
+
+
+internal struct ReactNativeSVGResource: SessionReplayResource {
+    let identifier: String
+    let svgContent: String
+    let mimeType: String = "image/svg+xml"
+
+    init(id: String, svgContent: String) {
+        self.identifier = id
+        self.svgContent = svgContent
+    }
+
+    func calculateIdentifier() -> String {
+        return identifier
+    }
+
+    func calculateData() -> Data {
+        return svgContent.data(using: .utf8) ?? Data()
+    }
+}
+
+
+internal class SvgViewRecorder: SessionReplayNodeRecorder {
+    internal var identifier = UUID()
+
+    internal let uiManager: RCTUIManager
+    internal let fabricWrapper: RCTFabricWrapper
+    internal let svgMap: [String: SVGData]
+
+    internal init(uiManager: RCTUIManager, fabricWrapper: RCTFabricWrapper, svgMap: [String: SVGData]) {
+        self.uiManager = uiManager
+        self.fabricWrapper = fabricWrapper
+        self.svgMap = svgMap
+    }
+    
+    func semantics(
+        of view: UIView,
+        with attributes: SessionReplayViewAttributes,
+        in context: SessionReplayViewTreeRecordingContext
+    ) -> SessionReplayNodeSemantics? {
+        
+        if (view.accessibilityIdentifier != nil) {
+            let viewId = context.ids.nodeID(view: view, nodeRecorder: self)
+            let subView = view.subviews[0]
```

Comment:
Can't this fail? Will the view always have at least one subview? In any case, it's safer to safely unwrap it. 

```suggestion
            guard let subView = view.subviews.first else { return }
```